handlers/viber/viber.go

```diff
@@ -96,7 +105,42 @@ func (h *handler) receiveEvent(ctx context.Context, channel courier.Channel, w h
 		return nil, handlers.WriteAndLogRequestIgnored(ctx, h, channel, w, r, "webhook valid")
 
 	case "conversation_started":
-		return nil, handlers.WriteAndLogRequestIgnored(ctx, h, channel, w, r, "ignored conversation start")
+
+		msgText := channel.StringConfigForKey(configViberWelcomeMessage, "")
+		if msgText == "" {
+			return nil, handlers.WriteAndLogRequestIgnored(ctx, h, channel, w, r, "ignored conversation start")
+		}
+
+		authToken := channel.StringConfigForKey(courier.ConfigAuthToken, "")
+
+		viberID := payload.User.ID
+		ContactName := payload.User.Name
+
+		// build the URN
+		urn, err := urns.NewURNFromParts(urns.ViberScheme, viberID, "", "")
+		if err != nil {
+			return nil, handlers.WriteAndLogRequestIgnored(ctx, h, channel, w, r, "ignored conversation start")
+		}
+		// build the channel event
+		channelEvent := h.Backend().NewChannelEvent(channel, courier.WelcomeMessage, urn).WithContactName(ContactName)
+
+		err = h.Backend().WriteChannelEvent(ctx, channelEvent)
+		if err != nil {
+			return nil, err
+		}
+		payload := welcomeMessagePayload{
```

Comment:
would move this and below into `writeWelcomeMessage Response` (don't need Viber, already in Viber package)